stdlib/public/SwiftShims/CoreFoundationShims.h

```diff
@@ -40,6 +40,13 @@ typedef unsigned char _swift_shims_Boolean;
 typedef __swift_uint8_t _swift_shims_UInt8;
 typedef __swift_uint32_t _swift_shims_CFStringEncoding;
 
+typedef struct __swift_shims_builtin_CFString {
+  const void * _Nonnull isa; // point to __CFConstantStringClassReference
+  unsigned long flags;
+  const __swift_uint8_t * _Nonnull str;
+  unsigned long length;
+} _swift_shims_builtin_CFString;
```

Comment:
Does it matter that this type is mismatched on Linux?  There are three different ABIs in play :-(:

Default (ObjC) (`-fcf-runtime-abi=objc`):
```
struct __NSConstantString_tag {
  const int *isa;
  int flags;
  const char *str;
  long length;
} __NSConstantString;
```

Swift 4.1, 4.2 (`-fcfc-runtime-abi=swift-4.1`, `-fcf-runtime-abi=swift-4.2`):
```
typedef struct __NSConstantString_tag {
  uintptr_t _cfisa;
  uintptr_t _swift_rc;
  _Atomic(uint64_t) _cfinfoa;
  const char *_ptr;
  uint32_t _length;
} __NSConstantString;
```

Swift 5.0 (`-fcf-runtime=swift-5.0`):
```
typedef struct __NSConstantString_tag {
  uintptr_t _cfisa;
  uintptr_t _swift_rc;
  _Atomic(uint64_t) _cfinfoa;
  const char *_ptr;
  uintptr_t _length;
} __NSConstantString;
```

Reply:
This is only intended for Darwin, but maybe there should be a guard here to make it explicit and prevent anyone from mistakenly using it more widely in the future.

Reply:
I *believe* the CF shims file is already only built on Darwin, but thank you for pointing this out, it's good to be aware of.

Reply:
If this is meant to be Darwin only, I think that we should do something to make that explicit (perhaps move the file) - as it is included in the Linux/Android/Windows SDKs.